cost/aws/elb/clb_unused/AWS_Delete_Unused_CLB.pt

```diff
@@ -0,0 +1,243 @@
+	
+  result "content"
+  code <<-EOS
+  var param_exclude_tags_lower=param_exclude_tags.toString().toLowerCase().split(',');
+  var content=[]
+  for(var i=0; i<ds_aws_clb_list_tags.length; i++){
+    clb=ds_aws_clb_list_tags[i]
+
+    // Constructing availabilityZones with comma separated to display in detail_template
+    var availabilityZones=clb['availabilityZones']
+    var s_availabilityZones=""
+    for(var j=0; j < availabilityZones.length; j++){
+      availabilityZone = availabilityZones[j]
+      s_availabilityZones = s_availabilityZones + ', '+ availabilityZone
+    }
+
+    // Constructing IntanceIds with comma separated to display in detail_template
+    var instancesList = clb['instancesList']
+    var s_instanceIds=""
+    for(var j=0; j < instancesList.length; j++){
+      instanceId = instancesList[j]
+      s_instanceIds = s_instanceIds + ', '+ instanceId
+    }
+
+    //Constructing Tags of individual CLB into key=value format. Check, if the tag present in entered param_exclude_tags, ignore the CLB if the tag matches/present.
+    var tags = clb['tags']
+    var isTagMatched=false
+    var tagKeyValue=""
+    for(var j=0; j < tags.length; j++){
+      tag = tags[j]
+      if(param_exclude_tags_lower.indexOf((tag['tagKey']+'='+tag['tagValue']).toLowerCase()) !== -1){
+        isTagMatched = true;
+      }
+      // Constructing tags with comma separated to display in detail_template
+      tagKeyValue = tagKeyValue + ', '+ tag['tagKey']+'='+tag['tagValue']
+    }
+
+    //If the CLB tag does not match with entered param_exclude_tags, then check if any instance exists in the CLB and check for instances health.
+    if(!(isTagMatched)){
+      clb_details={
+        loadBalancerName: clb['loadBalancerName'],
+        instanceId:(s_instanceIds.slice(2)),
+        availabilityZones:(s_availabilityZones.slice(2)),
+        tagKeyValue:(tagKeyValue.slice(2))
+      }
+      if(instancesList.length > 0){
+        var isInstanceNotHealthy=false
+        // Check if the CLB has no healthy instance with state and description matches for the values mentioned below
+		var state= "OutOfService"
+        var description= "Instance has failed at least the UnhealthyThreshold number of health checks consecutively."
+        for(var k=0; k < ds_aws_instances_health.length; k++){
+          instance= ds_aws_instances_health[k]
+          if(((instance['loadBalancerName']) == (clb['loadBalancerName'])) && (instance['state'] == state && instance['description'] == description)){
+            isInstanceNotHealthy=true
+          }
+        }
+        if(isInstanceNotHealthy){
+          content.push(clb_details)
+        }
+      }else{
+        content.push(clb_details)
+      }
+    }
+  }
+EOS
+end
+
+datasource "ds_unused_clb_map" do
+  run_script $js_aws_clb_filter_map, $ds_aws_clb_list_tags, $ds_aws_instances_health, $param_exclude_tags
+end
+
+policy "policy_delete_unused_CLB" do
```

Comment:
Line indentation is not uniform in this policy statement.

Reply:
Line indentation is uniform now in this policy statement..